### ISSUE TITLE
Using codesandbox as a local IDE

### ISSUE THREAD

**Issue by pyros2097** [codesandbox/codesandbox-client#580]
I'm thinking of a way I can package a local version of codesandbox editor so that any developer can just add that to package to their project and then it would run a local version of codesandbox for the current project. That way even your job of getting/downloading an IDE for coding is removed and everything for you to get started on the project lies within the package.json.

Is this even feasible using codesandbox? Does the editor/previewer/explorer rely to much on the backend api? Any ideas on how to go about it?

Basically I want to build something like this,
https://github.com/alm-tools/alm
But using codesandbox.

**Comment by CompuIves** [comment]
I find this idea really interesting, I also think it isn't **too** hard to do (don't mark my words!).

I'm working on moving all data of a sandbox to files, as soon as we completely moved over the the files as source of truth we can create multiple implementations of the CodeSandbox API. For example, we could swap it out with Dropbox API, or swap it out with S3 API. Or even swap it out with the local `fs` (which makes most sense for the local version). Another challenge is Web Worker support, is it possible to use web workers locally?

So the things we need to first is:

- Move all data to files
	- Sandbox Name
	- Sandbox Description
	- Sandbox Tags
	- Sandbox Template Definition (just a string)

Ehh, I think that's all the data? Easier than I expected.

Next thing is bundling to make it work locally. I believe we should wrap it in Electron. Would be cool to know how that goes.

I'm still not sure if moving to Electron is a good idea, but if you're motivated I'd love to help you out with the efforts!

**Comment by FernandoGOT** [comment]
+1

**Comment by katerlouis** [comment]
I'd love to use code sandbox in a standalone app where the "back functionality" on scroll doesn't fuck with you so much. 

Electron for the win!
defenitely +1!

**Comment by m1911star** [comment]
Hi, recently i came up with sandbox, tried it, and it's great! looking forward to deploy locally~

**Comment by jmarceli** [comment]
Do you have any plans for making an app out of codesandbox, so it can be used for a regular development like any other IDE/editor? Is it (theoretically) possible to distribute codesandbox as a standalone app?

**Comment by geek4go** [comment]
good job

**Comment by Seankps** [comment]
a year passed and no official response. Can we do this?

**Comment by mdarif** [comment]
+1

This would be one the best offering of codesandbox, rather bigger then the current one, using codesandbox as a local editor, which would remove the dependency on other IDE's

Of course keeping code sharing capability limited as we don't want to expose anything outside world, in a secure environment!

**Comment by alexluong** [comment]
Just wanna chiming in and say that if it's a standalone app, we can finally use `Cmd + Number` to switch code tabs instead of Chrome tabs!!

**Comment by huylv-teko** [comment]
Hope this launh soon :D

**Comment by brysgo** [comment]
@CompuIves - Are all the components open source? If so, I think this would be a fun undertaking.

**Comment by pyros2097** [comment]
If anyone wan't to know I found this,
https://www.theia-ide.org/
